crates/common/mqtt_channel/src/session.rs

```diff
@@ -26,10 +26,11 @@ pub async fn init_session(config: &Config) -> Result<(), MqttError> {
                     return Err(err);
                 };
                 let subscriptions = config.subscriptions.filters();
-                if subscriptions.is_empty() {
-                    break;
+                for subscription in subscriptions {
+                    mqtt_client
+                        .subscribe(subscription.path, config.subscriptions.qos)
+                        .await?;
                 }
-                mqtt_client.subscribe_many(subscriptions).await?;
```

Comment:
Same comments as for `Connection::open()` plus a really important difference.

Here, the case `_ => ()` actually silently discards any received messages, since there is no `Ok(Event::Incoming(Packet::Publish(msg)))` match case.

This means that the idea of creating an MQTT session with a `--init` option is fragile and even buggy if done after the very first installation. If called twice or once the mapper or the agent has already been launched, this command will consume all the pending messages preventing them to be processed.

=> We need to rethink the way MQTT daemons are started. They must not assume that all their peers are ready to consume messages and they must handle with extra care all init messages - as the initial software list request sent by the c8y mapper on start.

Reply:
Will revert.